### PR TITLE
Doc: document standard pickup definition

### PR DESCRIPTION
**Pull request by Miepee**

### CODE REVIEW

File: randovania/game_description/pickup/standard_pickup.py [hunk 1]
```diff
@@ -20,26 +20,86 @@
...
+
     progression: tuple[str, ...]
+    """
+    Defines item resources (as short names) that collecting this pickup provides.
+    If this tuple contains only one resource, then every collection will give that resource.
+    If it contains more than one, then every time the pickup will be collected, it will give the Nth resource.
+    """
+
     preferred_location_category: LocationCategory
+    """The category for the preferred location."""
```
Comment (henriquegemignani) [inline randovania/game_description/pickup/standard_pickup.py:48]: I hate this kind of docstring, it adds nothing but a string that can become outdated.

Please explain what that is.

File: randovania/game_description/pickup/standard_pickup.py [hunk 2]
```diff
@@ -20,26 +20,86 @@
 @dataclass(frozen=True)
 class StandardPickupDefinition(JsonDataclass, DataclassPostInitTypeCheck):
     game: RandovaniaGame = dataclasses.field(metadata={"init_from_extra": True})
+    """The game this pickup comes from."""
+
     name: str = dataclasses.field(metadata={"init_from_extra": True})
+    """The name of the pickup."""
+
     pickup_category: PickupCategory = dataclasses.field(metadata={"init_from_extra": True})
+    """The precise category for this pickup."""
```
Comment (henriquegemignani) [inline randovania/game_description/pickup/standard_pickup.py:29]: Please explain what is the precise category

File: randovania/game_description/pickup/standard_pickup.py [hunk 3]
```diff
@@ -20,26 +20,86 @@
...
+
     broad_category: PickupCategory = dataclasses.field(metadata={"init_from_extra": True})
+    """The broad category for this pickup."""
```
Comment (henriquegemignani) [inline randovania/game_description/pickup/standard_pickup.py:32]: Please explain what is the broad category

File: randovania/game_description/pickup/standard_pickup.py [hunk 4]
```diff
@@ -20,26 +20,86 @@
...
+
     hide_from_gui: bool = dataclasses.field(default=False, metadata=EXCLUDE_DEFAULT)
+    """Whether this pickup should be hidden in the GUI."""
+
     must_be_starting: bool = dataclasses.field(default=False, metadata=EXCLUDE_DEFAULT)
+    """Whether the pickup is required be a starting item."""
+
     original_locations: tuple[PickupIndex, ...] = dataclasses.field(default_factory=tuple, metadata=EXCLUDE_DEFAULT)
+    """
+    The index of the pickup (defined in the database) on where this pickup should be located if
```
Comment (henriquegemignani) [inline randovania/game_description/pickup/standard_pickup.py:85]: Docstring is written as if the field is not a list of indices.

**Review by henriquegemignani — CHANGES_REQUESTED** [review]

File: randovania/game_description/pickup/standard_pickup.py (hunk 4 above)
Comment (Miepee) [reply randovania/game_description/pickup/standard_pickup.py:85]: I didn't even see that this was a tuple, good catch.

File: randovania/game_description/pickup/standard_pickup.py (hunk 1 above)
Comment (Miepee) [reply randovania/game_description/pickup/standard_pickup.py:48]: Shouldn't the LocationCategory class itself explain on what that is?
Not opposed to it, it's just the way I usually have seen things documented.

Comment (henriquegemignani) [reply randovania/game_description/pickup/standard_pickup.py:48]: `LocationCategory` determines what kind of location a pickup node is. Don't see why it should explain why a standard pickup has a preference for some kind of category to me

**Comment by Miepee** [comment]
tried making the specified docstrings clearer

**Review by henriquegemignani — APPROVED** [review]

**Pull request merged**